apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts

```diff
@@ -0,0 +1,235 @@
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+const addSharingSetPopover = (resLocals) => {
+  return html`
+    <form name="sharing-set-create" method="POST">
+      <input type="hidden" name="__action" value="unsafe_sharing_set_create">
+      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}">
+
+      <div class="form-group mb-4">
+        <p class=form-text>
+          Enter the name of the sharing set you would like to create.
+        </p>
+      </div>
+      <div class=form-group>
+        <input class="form-control form-control-sm" type="text" name="sharing_set_name" required/>
+      <div>
+      <div class="text-right mt-4">
+        <button type="button" class="btn btn-secondary" onclick="$('#courseSharingSetAdd').popover('hide')">Cancel</button>
+        <button type="submit" class="btn btn-primary">Create Sharing Set</button>
+      </div>
+    </form>
+  `.toString();
+};
+
+const chooseSharingNamePopover = (resLocals) => {
```

Comment:
In PrairieTest at least, we've been erring towards modals instead of popovers for this kind of thing. @mwest1066 do we have any kind of general rule for when to use one over the other?

Reply:
cc @mwest1066

Reply:
Matt says go for a modal, the general rule being that if you need to know what element of the page that the dialog is about or linked to, use a popover because it stays anchored to the element on the page. Otherwise, use a modal. 

This button is unique in the sense that there will only ever be one of it on a given page, so it makes sense for it to be a modal.

Reply:
done.